src/Callbacks/callback.jl

```diff
@@ -1,3 +1,34 @@
+"""
+    StateDSS{} <: AbstractCallback
+
+Specifies the application of a DSS operation to Fields of 
+prognostic variables at user defined timestep interval. 
+Recommended timestep interval is 1.
+"""
+struct StateDSS{M <: AbstractModel, I <: Number} <: AbstractCallback
+    model::M
+    interval::I
+end
+function (F::StateDSS)(integrator)
+    Y = integrator.u
+    for i in propertynames(Y)
```

Comment:
@jiahe23 Is there a cleaner way of accessing all the state fields from the simulation object ?

Reply:
I don't know any better ways...